src/Symfony/Component/PropertyInfo/Tests/Extractor/PhpDocExtractorTest.php

```diff
@@ -409,6 +410,27 @@ public function constructorTypesProvider()
         ];
     }
 
+    /**
+     * @dataProvider unionTypesProvider
+     */
+    public function testExtractorUnionTypes(string $property, ?array $types)
+    {
+        $this->assertEquals($types, $this->extractor->getTypes('Symfony\Component\PropertyInfo\Tests\Fixtures\DummyUnionType', $property));
+    }
+
+    public function unionTypesProvider(): array
+    {
+        return [
+            ['a', [new Type(Type::BUILTIN_TYPE_STRING), new Type(Type::BUILTIN_TYPE_INT)]],
+            ['b', [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, [new Type(Type::BUILTIN_TYPE_INT)], [new Type(Type::BUILTIN_TYPE_STRING), new Type(Type::BUILTIN_TYPE_INT)])]],
+            ['c', [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, [new Type(Type::BUILTIN_TYPE_INT)], [new Type(Type::BUILTIN_TYPE_STRING), new Type(Type::BUILTIN_TYPE_INT)])]],
+            ['d', null], // union keys are not supported by phpdocumentor/reflection
+            ['e', null], // union keys are not supported by phpdocumentor/reflection
+            // ['f', null], // constant expressions are not supported
```

Comment:
Should be removed then

Reply:
Ah ok, yes I read it.

Lets keep it for now

Reply:
Damn, I just pushed. Should I restore it? Or is the notice here enough?